pkg/testgridanalysis/testgridconversion/to_raw_data.go

```diff
@@ -25,29 +25,58 @@ type SyntheticTestManager interface {
 	CreateSyntheticTests(rawJobResults testgridanalysisapi.RawData) []string
 }
 
+type MissingOverallError struct {
+	jobName string
+}
+
+func (m MissingOverallError) Error() string {
+	return fmt.Sprintf("missing Overall test in job %s", m.jobName)
+}
+
 type ProcessingOptions struct {
 	SyntheticTestManager SyntheticTestManager
 	StartDay             int
 	NumDays              int
 }
 
 // ProcessTestGridDataIntoRawJobResults returns the raw data and a list of warnings encountered processing the data.
-func (o ProcessingOptions) ProcessTestGridDataIntoRawJobResults(testGridJobInfo []testgridv1.JobDetails) (testgridanalysisapi.RawData, []string) {
+// returns the raw data and a list of warnings encountered processing the data.
+func (o ProcessingOptions) ProcessTestGridDataIntoRawJobResults(testGridJobInfo []testgridv1.JobDetails) (testgridanalysisapi.RawData, []string, []error) {
 	rawJobResults := testgridanalysisapi.RawData{JobResults: map[string]testgridanalysisapi.RawJobResult{}}
 
+	errs := []error{}
+
 	for _, jobDetails := range testGridJobInfo {
 		klog.V(2).Infof("processing test details for job %s\n", jobDetails.Name)
 		startCol, endCol := computeLookback(o.StartDay, o.NumDays, jobDetails.Timestamps)
-		processJobDetails(rawJobResults, jobDetails, startCol, endCol)
+		if err := processJobDetails(rawJobResults, jobDetails, startCol, endCol); err != nil {
+			errs = append(errs, err)
+		}
 	}
 
 	// now that we have all the JobRunResults, use them to create synthetic tests for install, upgrade, and infra
 	warnings := o.SyntheticTestManager.CreateSyntheticTests(rawJobResults)
 
-	return rawJobResults, warnings
+	return rawJobResults, warnings, errs
+}
+
+func hasOverallTest(job testgridv1.JobDetails) bool {
+	for _, test := range job.Tests {
+		if test.Name == overall {
+			return true
+		}
+	}
+
+	return false
 }
 
-func processJobDetails(rawJobResults testgridanalysisapi.RawData, job testgridv1.JobDetails, startCol, endCol int) {
+func processJobDetails(rawJobResults testgridanalysisapi.RawData, job testgridv1.JobDetails, startCol, endCol int) error {
+	// Do a first pass to make sure the job has an "Overall" test
+	// If not, return early so we don't taint the rest of the results
+	if !hasOverallTest(job) {
```

Comment:
i'm curious, is this tripping on any of the jobs we're currently analyzing?  (i would expect/hope not).

just want to be sure before we turn it on and start throwing errors